File: MovieProject/MovieProject.Web/Modules/ServerTypes/Northwind/OrderSubtotalsRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface OrderSubtotalsRow {
    OrderId?: number;
    Subtotal?: number;
}

export abstract class OrderSubtotalsRow {
    static readonly idProperty = 'OrderId';
    static readonly localTextPrefix = 'Northwind.OrderSubtotals';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<OrderSubtotalsRow>();
}

File: MovieProject/MovieProject.Web/Modules/Northwind/OrderSubtotals/OrderSubtotalsRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace MovieProject.Northwind;

[ConnectionKey("Northwind"), Module("Northwind"), TableName("Order Subtotals")]
[DisplayName("Order Subtotals"), InstanceName("Order Subtotals")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class OrderSubtotalsRow : Row<OrderSubtotalsRow.RowFields>, IIdRow
{
    [DisplayName("Order Id"), Column("OrderID"), NotNull, IdProperty]
    public int? OrderId { get => fields.OrderId[this]; set => fields.OrderId[this] = value; }

    [DisplayName("Subtotal"), Size(19), Scale(4)]
    public decimal? Subtotal { get => fields.Subtotal[this]; set => fields.Subtotal[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field OrderId;
        public DecimalField Subtotal;

    }
}